src/d3a/models/myco_matcher/__init__.py

```diff
@@ -9,19 +9,32 @@
 
 
 class MycoMatcher:
-    def init(self):
-        if ConstSettings.IAASettings.BID_OFFER_MATCH_TYPE == \
-                BidOfferMatchAlgoEnum.PAY_AS_BID.value:
-            self.match_algorithm = PayAsBidMatcher()
-        elif ConstSettings.IAASettings.BID_OFFER_MATCH_TYPE == \
-                BidOfferMatchAlgoEnum.PAY_AS_CLEAR.value:
-            self.match_algorithm = PayAsClearMatcher()
-        elif is_external_matching_enabled():
-            self.match_algorithm = ExternalMatcher()
-        else:
-            raise WrongMarketTypeException("Wrong market type setting flag "
-                                           f"{ConstSettings.IAASettings.MARKET_TYPE}")
+    def __init__(self):
+        self.match_algorithm = None
+
+    def reconfigure(self):
+        """Reconfigure the myco matcher properties at runtime."""
+        self.match_algorithm = self.get_matcher_algorithm()
```

Comment:
IMO this is counter intuitive. `self.get_matcher_algorithm()` should also be called when the class is activated. I understand that it does not work when the class is initialised, but I would propose to stickt with the naming scheme that we have for the 2 stage initialisation on d3a and call this method `activate`. This will make the call on simulation.py more understandable IMO.
If we really need a `reconfigure` function in the future, then IMO, it should stay here and the `activate` method should be added. I know they would do the same for now, but maybe not in the future. What do you think?

Reply:
You are right, at the moment AFAIU the user cannot change the matching algorithm while a simulation is running, so this should be renamed to `activate`, and if later on we add the ability to change the algorithm we can add the reconfigure method